lib/utils.coffee

```diff
@@ -306,6 +309,12 @@ exports.requestAsync = (options) ->
 			headers: options.headers
 			uri: urlLib.parse(url)
 		return response
+	.catch (error) ->
```

Comment:
This can be
```coffee
.catch (-> retriesRemaining > 0), ->
	exports.requestAsync(options, retriesRemaining - 1)
```
or
```
p = p.then ...
if retriesRemaining > 0
	p = p.catch ->
		exports.requestAsync(options, retriesRemaining - 1)
return p
```
to avoid needing to manually rethrow